### PR TITLE
[Publication] jQuery usage cleaning

### PR DESCRIPTION
**Pull request by laemtl**
First attempt to clean progressively jQuery from the modules.
Replace $.ajax by fetch and $.ready by document.addEventListener('DOMContentLoaded')

Makes also use of arrow functions to avoid the use of bind(this)

* Will make a step toward #4213 resolution
* Testing requires the changes made in #7284

### CODE REVIEW

File: modules/publication/jsx/viewProject.js [hunk 1]
```diff
@@ -87,60 +79,66 @@ class ViewProject extends React.Component {
    */
   fetchData() {
     let self = this;
```
Comment (maltheism) [inline modules/publication/jsx/viewProject.js:81]: Hi @laemtl, I believe `let self = this` isn't necessary with your changes.

**Review by maltheism — CHANGES_REQUESTED** [review]

File: modules/publication/jsx/viewProject.js (hunk 1 above)
Comment (laemtl) [reply modules/publication/jsx/viewProject.js:81]: Your totally right, thanks for catching this!

Comment (laemtl) [reply modules/publication/jsx/viewProject.js]: @maltheism Fixed.

**Review by maltheism — APPROVED** [review]

**Comment by driusan** [comment]
Seems to have conflicts..

**Comment by driusan** [comment]
Skimmed it and it looks good (if long overdue!) but it has conflicts and can't be merged

**Comment by laemtl** [comment]
@driusan Rebased.

**Pull request merged**